core/translate/planner.rs

```diff
@@ -0,0 +1,359 @@
+            };
+            let alias = maybe_alias
+                .map(|a| match a {
+                    ast::As::As(id) => id,
+                    ast::As::Elided(id) => id,
+                })
+                .map(|a| a.0);
+
+            (table, alias.unwrap_or(qualified_name.name.0))
+        }
+        _ => todo!(),
+    };
+
+    let mut node = Operator::Scan {
+        table: first_table.0.clone(),
+        predicates: None,
+        table_identifier: first_table.1.clone(),
+        id: node_id_counter.get_next_id(),
+    };
+
+    let mut tables = vec![first_table];
+
+    for join in from.joins.unwrap_or_default().into_iter() {
+        let (right, outer, predicates) = parse_join(schema, join, node_id_counter, &mut tables)?;
+        node = Operator::Join {
+            left: Box::new(node),
+            right: Box::new(right),
+            predicates,
+            outer,
+            id: node_id_counter.get_next_id(),
+        }
+    }
+
+    return Ok((node, tables));
+}
+
+fn parse_join(
+    schema: &Schema,
+    join: ast::JoinedSelectTable,
+    node_id_counter: &mut NodeIdCounter,
+    tables: &mut Vec<(Rc<BTreeTable>, String)>,
+) -> Result<(Operator, bool, Option<Vec<ast::Expr>>)> {
+    let ast::JoinedSelectTable {
+        operator,
+        table,
+        constraint,
+    } = join;
+
+    let table = match table {
+        ast::SelectTable::Table(qualified_name, maybe_alias, _) => {
+            let Some(table) = schema.get_table(&qualified_name.name.0) else {
+                crate::bail_parse_error!("Table {} not found", qualified_name.name.0);
+            };
+            let alias = maybe_alias
+                .map(|a| match a {
+                    ast::As::As(id) => id,
+                    ast::As::Elided(id) => id,
+                })
+                .map(|a| a.0);
+
+            (table, alias.unwrap_or(qualified_name.name.0))
+        }
+        _ => todo!(),
+    };
+
+    tables.push(table.clone());
+
+    let outer = match operator {
+        ast::JoinOperator::TypedJoin(Some(join_type)) => {
+            if join_type == JoinType::LEFT | JoinType::OUTER {
+                true
+            } else if join_type == JoinType::RIGHT | JoinType::OUTER {
+                true
+            } else {
+                false
+            }
+        }
+        _ => false,
+    };
+
+    let predicates = constraint.map(|c| match c {
+        ast::JoinConstraint::On(expr) => break_predicate_at_and_boundaries(expr, vec![]),
+        ast::JoinConstraint::Using(_) => todo!("USING joins not supported yet"),
+    });
+
+    Ok((
+        Operator::Scan {
+            table: table.0.clone(),
+            predicates: None,
+            table_identifier: table.1.clone(),
+            id: node_id_counter.get_next_id(),
+        },
+        outer,
+        predicates,
+    ))
+}
+
+fn break_predicate_at_and_boundaries(
+    predicate: ast::Expr,
+    mut predicates: Vec<ast::Expr>,
+) -> Vec<ast::Expr> {
+    match predicate {
```

Comment:
```suggestion
    out_predicates: &mut Vec<ast::Expr>,
) {
    match predicate {
```
```suggestion
) -> Vec<ast::Expr> {
    let mut out_predicates: Vec<ast::Expr>;
    match predicate {
```

Reply:
nit

Reply:
sorta implemented this, made the function take in a mutable reference to `out_predicates`